FreeAPSWatch WatchKit Extension/Views/NutrientsView.swift

```diff
@@ -0,0 +1,192 @@
+                )
+
+                GramsPicker(
+                    selection: $proteinCount,
+                    label: "Protein",
+                    labelAlignment: .center,
+                    max: defaultMaxProtein,
+                    accentColor: .red
+                )
+
+                GramsPicker(
+                    selection: $fatCount,
+                    label: "Fat",
+                    labelAlignment: .trailing,
+                    max: defaultMaxFat,
+                    accentColor: .orange
+                )
+            }
+            .padding(.bottom, 24)
+            HStack {
+                Button {
+                    WKInterfaceDevice.current().play(.click)
+                    watchStateModel.isCarbsViewActive = false
+                } label: {
+                    Text("Cancel")
+                }
+                Button {
+                    WKInterfaceDevice.current().play(.click)
+                    watchStateModel.addNutrients([
+                        .carb: carbCount,
+                        .protein: proteinCount,
+                        .fat: fatCount
+                    ])
+                } label: {
+                    Text("Add").opacity(isSubmitButtonDisabled ? 0.4 : 1)
+                }
+                .tint(.loopYellow)
+                .disabled(isSubmitButtonDisabled)
+            }
+            .padding([.leading, .trailing], 4)
+            .padding(.bottom, 12)
+        }
+        .frame(maxHeight: .infinity)
+        .edgesIgnoringSafeArea(.bottom)
+        .navigationTitle("Nutrients")
+        .onAppear {
+            if let unwrappedCarbsRequired = watchStateModel.carbsRequired {
+                carbCount = Int(unwrappedCarbsRequired)
+            }
+        }
+    }
+
+    struct GramsPicker: View {
+        @Environment(\.sizeCategory) var sizeCategory
+        @State var step = Constants.defaultPickerStep
+        let labelFrameWidth = WKInterfaceDevice.current().screenBounds.width - 8
+
+        @Binding var selection: Int
+        let label: String
+        let labelAlignment: Alignment
+        let max: Int
+        let accentColor: Color
+
+        var fontForSizeCategory: Font {
+            switch sizeCategory {
+            case .extraSmall,
+                 .small:
+                return .system(size: 12)
+            case .medium:
+                return .system(size: 14)
+            case .large:
+                return .system(size: 16)
+            default:
+                return .system(size: 20)
+            }
+        }
+
+        var fontForSizeCategoryForHundreds: Font {
+            switch sizeCategory {
+            case .extraSmall:
+                return .system(size: 8)
+            case .small:
+                return .system(size: 9)
+            case .large,
+                 .medium:
+                return .system(size: 14)
+            default:
+                return .system(size: 16)
+            }
+        }
```

Comment:
Numbers below 100 will use `fontForSizeCategory` scale, and all above 99 will use `fontForSizeCategoryForHundreds` accordingly. Tweaked for different sizes: 38, 40, 41, 44, 45 starting from Apple Watch Series 3.

Reply:
Removed.